assets/scripts/auth/ui.js

```diff
@@ -37,10 +41,12 @@ const onChangePwFailure = function (response) {
 }
 
 const onSignOutSuccess = function (response) {
-  $('#sign-up-message').text('sign out success')
-  $('.sign-in-wrapper').removeClass('hidden')
-  $('.main-content').addClass('hidden')
-  $('#change-pw').trigger('reset')
+  store.user = null // wipes signed-in users data clean
+  $('.modal').modal('hide')
+  $('#show-modal-body').empty()
```

Comment:
I'm pretty sure the issue you're trying to solve here is in `photo/ui.js`. You're clearing the show-modal-body alright, but `onShowPhotoSuccess` is setting the html of ALL elements with class `modal-content` (including the sign-in modal).

This was fine when we only had one modal, but at this point we need to change it to only set the html of the show-photo modal and leave the sign in/up ones alone.

Reply:
If you do end up needing to mess around with other ways of hiding/clearing modals or their content though, you could look at either other modal methods:
https://getbootstrap.com/docs/4.0/components/modal/#methods

or some other jQuery methods, like .remove() or .detach():
https://www.w3schools.com/jquery/html_detach.asp